packages/react-components/src/components/FileCard.tsx

```diff
@@ -44,15 +46,42 @@ export interface _FileCardProps {
   actionHandler?: () => void;
 }
 
+/**
+ * Strings of {@link _FileCard} that can be overridden.
+ *
+ * @public
+ */
+export interface FileCardStrings {
+  /** Aria label to notify user about the progress of file upload. */
+  uploadProgress?: string;
+  /** Aria label to notify user that the file upload started. */
+  uploading?: string;
+  /** Aria label to notify user that the file upload completed. */
+  uploadCompleted?: string;
+}
+
 /**
  * @internal
  * A component for displaying a file card with file icon and progress bar.
  */
 export const _FileCard = (props: _FileCardProps): JSX.Element => {
   const { fileName, fileExtension, progress, actionIcon } = props;
   const theme = useTheme();
+  const localeStrings = useLocale().strings.fileCard;
+  const [announcerString, setAnnouncerString] = useState<string | undefined>(undefined);
 
   const showProgressIndicator = progress !== undefined && progress > 0 && progress < 1;
+
+  useEffect(() => {
+    if (showProgressIndicator) {
+      setAnnouncerString(`${localeStrings.uploading} ${fileName}`);
+    } else if (progress && progress >= 1) {
```

Comment:
progress can never be > 1, not sure why we are checking that here.

Reply:
This can be simplified. `if (progress === 1)`

Reply:
done